app/javascript/components/server-components/BalancePage.tsx

```diff
@@ -598,14 +600,266 @@ const PeriodPaypalAccount = ({
+        </p>
+
+        <fieldset>
+          <label htmlFor="instant-payout-date">Pay out balance up to</label>
+          <div className="input cursor-pointer">
+            <Icon name="calendar-all" />
+            <select
+              id="instant-payout-date"
+              value={instantPayoutId}
+              onChange={(e) => setInstantPayoutId(e.target.value)}
+            >
+              {instant_payout.payable_balances.map((balance) => (
+                <option key={balance.id} value={balance.id}>
+                  {new Date(balance.date).toLocaleDateString()}
+                </option>
+              ))}
+            </select>
+            <Icon name="outline-cheveron-down" />
+          </div>
+        </fieldset>
+        <fieldset>
+          <legend>Payout details</legend>
+          <div className="cart">
+            <div className="cart-summary">
+              <div>
+                <p>Sent to</p>
+                <div>
+                  {instant_payout.bank_account_type === "CARD" ? (
+                    <p>
+                      <span>
+                        {instant_payout.routing_number} {instant_payout.account_number}
+                      </span>
+                    </p>
+                  ) : (
+                    <div>
+                      {instant_payout.bank_name ? <p className="text-right">{instant_payout.bank_name}</p> : null}
+                      <p className="text-right">
+                        Routing number: <span>{instant_payout.routing_number}</span>
+                      </p>
+                      <p className="text-right">
+                        Account: <span>{instant_payout.account_number}</span>
+                      </p>
+                    </div>
+                  )}
+                </div>
+              </div>
+              <div>
+                <p>Amount</p>
+                <div>${formatPriceCentsWithoutCurrencySymbol("usd", instantPayoutAmountCents)}</div>
+              </div>
+              <div>
+                <p>Instant payout fee ({INSTANT_PAYOUT_FEE_PERCENTAGE * 100}%)</p>
+                <div>
+                  -$
+                  {formatPriceCentsWithoutCurrencySymbol("usd", instantPayoutFee)}
+                </div>
+              </div>
+            </div>
+            <footer>
+              <p>
+                <strong>You'll receive</strong>
+              </p>
+              <div>${formatPriceCentsWithoutCurrencySymbol("usd", instantPayoutAmountCents - instantPayoutFee)}</div>
+            </footer>
+          </div>
+          {instantPayoutAmountCents > MAXIMUM_INSTANT_PAYOUT_AMOUNT_CENTS ? (
+            <div role="status" className="info">
+              Your balance exceeds the maximum amount for a single instant payout, so we'll automatically split your
+              balance into multiple payouts.
+            </div>
+          ) : null}
+        </fieldset>
+      </Modal>
+    </div>
+  );
+};
+
+const DailyInstantPeriod = ({
```

Comment:
This is a new component for users with daily frequency set.

We do not have an easy mechanism to show unpaid balances for the next daily payment period (from UTC 08:00 on the current day to UTC 08:00 on the next day). So I have avoided showing a balance for now.

Once the daily payouts happen they will show up in past payments.

In the future, we should completely refactor `next_payout_period_data` and unify the logic on this page.